### PR TITLE
Feature/sendbuffer plus some refactoring

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: protocols/bgp/packet/path_attributes_test.go [hunk 1]
```diff
@@ -1582,14 +1582,14 @@ func TestSerialize(t *testing.T) {
 func TestSerializeAddPath(t *testing.T) {
 	tests := []struct {
 		name     string
-		msg      *BGPUpdateAddPath
+		msg      *BGPUpdate
 		expected []byte
 		wantFail bool
 	}{
-		{
+		/*{
```
Comment (czerwonk) [inline protocols/bgp/packet/path_attributes_test.go]: please uncomment or remove

File: protocols/bgp/server/withdraw_test.go [hunk 2]
```diff
@@ -111,7 +111,9 @@ func TestWithDrawPrefixesAddPath(t *testing.T) {
 	}
 	for _, tc := range testcases {
 		buf := bytes.NewBuffer([]byte{})
-		opt := &packet.Options{}
+		opt := &types.Options{
+			AddPathRX: true,
+		}
 		err := withDrawPrefixesAddPath(buf, opt, tc.Prefix, tc.Path)
```
Comment (czerwonk) [inline protocols/bgp/server/withdraw_test.go]: we should consider renaming it to withdrawPrefixes and let this method decide which type of withdraw message is needed internaly based on the options passed

File: protocols/bgp/types/options.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package types
+
+// Options represents options to the update sender
```
Comment (czerwonk) [inline protocols/bgp/types/options.go]: as well as the decoder and encoder

File: protocols/bgp/server/withdraw_test.go (hunk 2 above)
Comment (czerwonk) [reply protocols/bgp/server/withdraw_test.go]: nothing to do here for now

File: protocols/bgp/packet/path_attributes_test.go (hunk 1 above)
Comment (taktv6) [reply protocols/bgp/packet/path_attributes_test.go]: Fixed.

File: protocols/bgp/types/options.go (hunk 3 above)
Comment (taktv6) [reply protocols/bgp/types/options.go]: Fixed.

**Pull request merged**